test/extended/prometheus/collection_profiles.go

```diff
@@ -0,0 +1,365 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	helper "github.com/openshift/origin/test/extended/util/prometheus"
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	prometheusoperatorv1client "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned/typed/monitoring/v1"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"gopkg.in/yaml.v3"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	projectName = "monitoring-collection-profiles"
+
+	collectionProfileFeatureLabel = "monitoring.openshift.io/collection-profile"
+	collectionProfileFull         = "full"
+	collectionProfileDefault      = collectionProfileFull
+	collectionProfileMinimal      = "minimal"
+	collectionProfileNone         = ""
+
+	operatorName              = "cluster-monitoring-operator"
+	operatorNamespaceName     = "openshift-monitoring"
+	operatorConfigurationName = "cluster-monitoring-config"
+)
+
+var (
+	oc   = exutil.NewCLI(projectName)
+	tctx = context.Background()
+
+	collectionProfilesSupportedList = []string{
+		collectionProfileFull,
+		collectionProfileMinimal,
+	}
+)
+
+type runner struct {
+	kclient                       kubernetes.Interface
+	mclient                       *prometheusoperatorv1client.MonitoringV1Client
+	pclient                       prometheusv1.API
+	originalOperatorConfiguration *v1.ConfigMap
+}
+
+// NOTE: The nested `Context` containers inside the following `Describe` container are used to group certain tests based on the environments they demand.
+// NOTE: When adding a test-case, ensure that the test-case is placed in the appropriate `Context` container.
+// NOTE: The containers themselves are guaranteed to run in the order in which they appear.
+var _ = g.Describe("[sig-instrumentation][OCPFeatureGate:MetricsCollectionProfiles] The collection profiles feature-set", g.Ordered, func() {
+	defer g.GinkgoRecover()
+
+	o.SetDefaultEventuallyTimeout(15 * time.Minute)
+	o.SetDefaultEventuallyPollingInterval(5 * time.Second)
+
+	r := &runner{}
+
+	g.BeforeAll(func() {
+		if !exutil.IsTechPreviewNoUpgrade(tctx, oc.AdminConfigClient()) {
```

Comment:
Signature changed during rebase, so modified this (from `exutil.IsTechPreviewNoUpgrade(oc)`.